### PR TITLE
Prepare pip package

### PR DESCRIPTION
**Pull request by florian-huber**
- made ``scipy`` and ``rdkit`` optional imports
- add requirements to ``setup.py``
- add CI to test pip package ~~--> does not work with ``macos-latest`` because that will install Python 2, no idea why.~~
- CI upon ``[release]`` still needs added token to secrets (I will do the first pypi upload manually and then add the token to this repo)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -43,15 +43,40 @@
...
+        "pyteomics >=4.2",
+        "pytest",
+        "pytest-cov",
+        "pyyaml",
+        "recommonmark",
+        "requests",
+        "sphinx>=3.0.0,!=3.2.0,<4.0.0",
+        "sphinx_rtd_theme",
+        "sphinxcontrib-apidoc",
+        "yapf",
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply setup.py]: Also the list of dev deps is repeating the install_requires which it should not and it has libs that are not used like `lxml`. See #149 for conda unused deps

File: tests/test_add_fingerprint.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import numpy
+import pytest
 from matchms import Spectrum
 from matchms.filtering import add_fingerprint
 
 
 def test_add_fingerprint_from_smiles():
     """Test if fingerprint it generated correctly."""
+    pytest.importorskip("rdkit")
```
Comment (sverhoeven) [inline tests/test_add_fingerprint.py:9]: Nice find, did not know pytest had this functionality.

File: .github/workflows/CI_push_pypi.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Python info
+        shell: bash -l {0}
+        run: |
+          which python
+          python --version
```
Comment (sverhoeven) [inline .github/workflows/CI_push_pypi.yml:24]: For MacOS failures, some OS have symlink of python -> python2 other OSes have python -> python3
```suggestion
          python3 --version
```

or maybe removing shell will help it pick right python

**Review by sverhoeven — CHANGES_REQUESTED** [review]
In `/README.rst`can you add docs for installation using pip and it's shortcoming.

File: .github/workflows/CI_push_pypi.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        run: |
+          python setup.py build
+      - name: Show pip list
+        shell: bash -l {0}
+        run: |
+          pip list
+      - name: Test
+        shell: bash -l {0}
+        run: |
+          python setup.py test
```
[the remark this answers is not in the document]
Comment (florian-huber) [reply .github/workflows/CI_push_pypi.yml]: Thanks, didn't know that.

File: matchms/similarity/CosineHungarian.py [hunk 5]
```diff
@@ -1,12 +1,20 @@
...
 
+try:  # scipy is not included in pip package (only matchs["cosine_hungarian"])
+    from scipy.optimize import linear_sum_assignment
+except ImportError:
+    _has_scipy = False
+else:
+    _has_scipy = True
+scipy_missing_message = "Package 'scipy' is required for this functionality."
+
+
```
[the remark this answers is not in the document]
Comment (florian-huber) [reply matchms/similarity/CosineHungarian.py]: It's true, it is simple to install so we could keep it non-optional. My thinking here was that ``CosineHungarian`` is indeed a reference similarity function that most people won't use in practice, and it is the only reason we would have ``scipy`` installed in matchms, and thereby also in spec2vec. But if that feels weird, we can keep it.

File: setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (florian-huber) [reply setup.py]: Thanks, wasn't aware of this! I have chosen the "dev" in ``extras_require`` option.

File: .github/workflows/CI_push_pypi.yml (hunk 3 above)
Comment (florian-huber) [reply .github/workflows/CI_push_pypi.yml:24]: Thanks Stefan! I removed the ``shell`` part which indeed fixed the macos issue I had.

File: tests/test_utils.py [hunk 6]
```diff
@@ -133,3 +160,36 @@ def test_derive_fingerprint_different_types_from_smiles():
...
         fingerprint = derive_fingerprint_from_smiles("[C+]#C[O-]", fingerprint_type, 16)
         assert numpy.all(fingerprint == expected_fingerprints[i]), "Expected different fingerprint."
+
+
+def test_missing_rdkit_module_error():
+    """Test if different functions return correct error when *rdkit* is not available"""
+    pytest.importorskip("rdkit")
+    expected_msg = "Conda package 'rdkit' is required for this functionality."
+
+    with mock.patch.dict(sys.modules, {"rdkit": None}):
```
[the remark this answers is not in the document]
Comment (florian-huber) [reply tests/test_utils.py]: Added this, but pylint doesn't like "Import outside toplevel" and unused imports (Chem). 
I moved it to top level as a workaround.

**Comment by florian-huber** [comment]
Thanks a lot for the comment so far @sverhoeven .
I still want to clean up the workflows a bit more (maybe fuse them to use the ``needs:`` field), but I made a separate issue on that #150.
Also let me know what you think of making ``scipy`` optional (mostly because of having spec2vec in mind) or not.

**Comment by sverhoeven** [comment]
I would keep scipy as an dependency. Less surprises and easier code.

**Review by sverhoeven — APPROVED** [review]

File: conda/environment-dev.yml [hunk 7]
```diff
@@ -23,8 +23,6 @@ dependencies:
     - pytest
     - pytest-cov
     - pytest-runner
-    - recommonmark
```
Comment (sverhoeven) [inline conda/environment-dev.yml:26]: The dependency list can be minimized to
```
dependencies:
  - python >=3.7,<3.9
  - pip
  - rdkit >=2020.03.1
  - pip:
    - ..[dev]
```
So we dont repeat the dev deps.

Comment (florian-huber) [reply conda/environment-dev.yml:26]: Good idea! I'll try that.

File: matchms/similarity/CosineHungarian.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (florian-huber) [reply matchms/similarity/CosineHungarian.py]: Scipy is not optional anymore.

[the remark this answers is not in the document]
Comment (sverhoeven) [reply matchms/similarity/CosineHungarian.py]: great

File: conda/environment-dev.yml (hunk 7 above)
Comment (florian-huber) [reply conda/environment-dev.yml:26]: Gave some issues with ``python setup.py test`` which I now changed to ``pytest``.

**Comment by florian-huber** [comment]
Thanks a lot for reviewing Stefan! 
I will later upload matchms to pypi and then setup a similar thing for spec2vec.

**Pull request merged**